library/std/src/sys/pal/unix/thread.rs

```diff
@@ -129,25 +129,28 @@ impl Thread {
         }
     }
 
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "dragonfly"))]
     pub fn set_name(name: &CStr) {
-        const TASK_COMM_LEN: usize = 16;
-
         unsafe {
-            // Available since glibc 2.12, musl 1.1.16, and uClibc 1.0.20.
-            let name = truncate_cstr::<{ TASK_COMM_LEN }>(name);
+            let name = {
+                cfg_if::cfg_if! {
+                    if #[cfg(target_os = "linux")] {
+                        const TASK_COMM_LEN: usize = 16;
+                        &truncate_cstr::<{ TASK_COMM_LEN }>(name)
+                    } else {
+                        name.to_bytes()
+                    }
+                }
+            };
```

Comment:
Minor nit to get rid of one level of nesting

```suggestion
            cfg_if::cfg_if! {
                if #[cfg(target_os = "linux")] {
                    const TASK_COMM_LEN: usize = 16;
                    let name = &truncate_cstr::<{ TASK_COMM_LEN }>(name);
                } else {
                    let name = name.to_bytes();
                }
            }
```

Otherwise lgtm, could you please squash?